### PR TITLE
Allow voluntary exit command to write to file rather than publish

### PR DESCRIPTION
**Pull request by rolfyone**
The advantage of this is that the message can avoid epoch validation, and also it can separate the submission of the exit from creation, so if something fails, it can have further investigation.

The format saved to json will allow it to be posted directly to the voluntary exits beacon-api POST endpoint.

Assists with #8158 but only as a potential workaround, allowing us to avoid setting up the keymanager-api to generate an exit.

## Documentation

- [X] I thought about documentation and added the `doc-change-required` label to this PR if updates are required.

## Changelog

- [X] I thought about adding a changelog entry, and added one if I deemed necessary.

### CODE REVIEW

File: teku/src/main/java/tech/pegasys/teku/cli/subcommand/VoluntaryExitCommand.java [hunk 1]
```diff
@@ -180,6 +206,30 @@ public Integer call() {
...
+              "%s exists and is not a directory, cannot export to this path.",
+              voluntaryExitsFolder));
+      return false;
+    } else if (!voluntaryExitsFolder.exists()) {
+      voluntaryExitsFolder.mkdirs();
+    }
+    final AtomicInteger failures = new AtomicInteger();
+    getValidatorIndices(validatorsMap)
+        .forEach(
+            (k, v) -> {
```
Comment (tbenr) [inline teku/src/main/java/tech/pegasys/teku/cli/subcommand/VoluntaryExitCommand.java]: I'd call this publicKey and validatorIndex for readability

File: teku/src/main/java/tech/pegasys/teku/cli/subcommand/VoluntaryExitCommand.java [hunk 2]
```diff
@@ -261,6 +305,43 @@ private void submitExitForValidator(final BLSPublicKey publicKey, final int vali
...
+        Optional.ofNullable(validatorsMap.get(publicKey))
+            .orElseThrow()
+            .getSigner()
+            .signVoluntaryExit(message, forkInfo)
+            .join();
+    return new tech.pegasys.teku.spec.datastructures.operations.SignedVoluntaryExit(
+        message, signature);
+  }
+
+  private boolean storeExitForValidator(BLSPublicKey blsPublicKey, Integer validatorIndex) {
```
Comment (tbenr) [inline teku/src/main/java/tech/pegasys/teku/cli/subcommand/VoluntaryExitCommand.java]: finals

File: teku/src/main/java/tech/pegasys/teku/cli/subcommand/VoluntaryExitCommand.java [hunk 3]
```diff
@@ -261,6 +305,43 @@ private void submitExitForValidator(final BLSPublicKey publicKey, final int vali
...
+          voluntaryExitsFolder.toPath().resolve(blsPublicKey.toAbbreviatedString() + "_exit.json"),
+          prettyExitMessage(exit));
+      return true;
+    } catch (IOException e) {
+      SUB_COMMAND_LOG.error("Failed to store exit for " + blsPublicKey.toAbbreviatedString());
+      return false;
+    }
+  }
+
+  private String prettyExitMessage(
```
Comment (tbenr) [inline teku/src/main/java/tech/pegasys/teku/cli/subcommand/VoluntaryExitCommand.java:339]: to be an actual prettyPrint you should use something similar to:

https://github.com/ConsenSys/teku/blob/8c172a2eaff954a73a5f665828bf17bdf04c1b6f/teku/src/main/java/tech/pegasys/teku/cli/subcommand/debug/PrettyPrintCommand.java#L109

essentially calling `gen.useDefaultPrettyPrinter()`

**Review by tbenr — APPROVED** [review]
Just some nits and a prettyPrint suggestion

**Review by lucassaldanha — APPROVED** [review]
LGTM. Just need a follow-up docs PR.

File: teku/src/main/java/tech/pegasys/teku/cli/subcommand/VoluntaryExitCommand.java (hunk 2 above)
Comment (rolfyone) [reply teku/src/main/java/tech/pegasys/teku/cli/subcommand/VoluntaryExitCommand.java]: its so annoying generated interfaces miss that :/ i got most :/

File: teku/src/main/java/tech/pegasys/teku/cli/subcommand/VoluntaryExitCommand.java (hunk 3 above)
Comment (rolfyone) [reply teku/src/main/java/tech/pegasys/teku/cli/subcommand/VoluntaryExitCommand.java:339]: fair, i was probably lazy about this a little.

**Pull request merged**